### DISCUSSION TITLE
Starting point for a fanuc S420F arm

### DISCUSSION THREAD

**Discussion by freemoore** [visose/Robots#280] [Q&A] [answered]
Hi all. I've just found this grasshopper plugin. It looks like I'll need to develop my own library as my robot arm is a Fanuc S420F, not listed. It has a 4-bar parallelogram linkage between J2 and J3 (where J1 is the base and J6 is the end effector). This means that if only J2 motor moves, both J2 and J3 angles change such that J3 maintains its angle relative to the base, rather than to J2 as you'd expect of straightforward serial kinematics. 

Are any of the existing robots in the library of a similar construction? I can change an existing library by entering the relevant DH parameters, and can model the physical shell of the machine, but I'm not sure how to make a new kinematics module. Any pointers appreciated, thanks!

**Comment by visose** [comment, marked as answer]
Hi, I presume you can still simulate the main joints of this robot with the same kinematics solver used by the other Fanuc robots. You probably don't really need to accurately simulate how this extra link moves when the main joints move. You don't need it to check if the robot will be able to reach your targets, and unlinkely that you need it to check if there will be a collision. If you want to simulate it, it would purely for visual purposes.
To simulate it, you can create the robot library without the geometry of the extra link, and then use the Planes output of the Simulation component and standard Grasshopper components to orient the geometry of this link into the right position. When the main joints move, the Planes will change and the link geometry should re-orient relative to the joints.
